Core/include/Acts/Propagator/EigenStepper.hpp

```diff
@@ -242,17 +242,18 @@ class EigenStepper {
   ///
   /// @param [in,out] state The stepping state (thread-local cache)
   /// @param [in] surface The surface provided
+  /// @param [in] index The surface intersection index
   /// @param [in] navDir The navigation direction
   /// @param [in] bcheck The boundary check for this status update
   /// @param [in] surfaceTolerance Surface tolerance used for intersection
   /// @param [in] logger A @c Logger instance
   Intersection3D::Status updateSurfaceStatus(
```

Comment:
I agree this function is mainly used to update an existing intersection. The problem is that we currently not always have an intersection object in hand while calling this function and I wanted to avoid changing the calling code too much

Reply:
I tried for 15 min and it seems to be a nightmare... for example in the navigator when we update the status for the target surface we do not have an intersection at hand and it seems bad to create one just to pass in an index

Reply:
That's why I was thinking it could be both, where we try to use the intersection overload and if we don't have an intersection we use the other one. But this is not critical for right now I think.